src/gameboy/mod.rs

```diff
@@ -75,12 +76,29 @@ impl Gameboy {
 
         'running: loop {
             self.handle_sdl2_events()?;
+            self.handle_interrupts();
             self.cpu.step();
         }
 
         Ok(())
     }
 
+    fn handle_interrupts(&mut self) {
+        for interrupt in Interrupt::iterator() {
+            let mut should_trigger = false;
+
+            {
+                let intc = &self.mmu.borrow().intc;
+                should_trigger = intc.should_trigger(interrupt);
+            }
+
+            if should_trigger {
+                println!("triggering {:?}", interrupt);
+                self.cpu.trigger_interrupt(interrupt);
```

Comment:
Or perhaps here is where we should disable the interrupt. Need to clear the flag also